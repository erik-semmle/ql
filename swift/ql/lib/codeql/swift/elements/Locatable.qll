private import codeql.swift.generated.Locatable
private import codeql.swift.elements.File

class Locatable extends LocatableBase {
  pragma[nomagic]
  override Location getLocation() {
    result = LocatableBase.super.getLocation()
    or
    not exists(LocatableBase.super.getLocation()) and result instanceof UnknownLocation
  }

  /**
   * Gets the primary file where this element occurs.
   */
  File getFile() { result = getLocation().getFile() }
}
