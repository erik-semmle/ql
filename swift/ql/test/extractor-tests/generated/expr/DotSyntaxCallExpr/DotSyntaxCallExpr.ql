// generated by codegen/codegen.py
import codeql.swift.elements
import TestUtils

from DotSyntaxCallExpr x, Expr getFunction, Expr getBaseExpr
where
  toBeTested(x) and
  not x.isUnknown() and
  getFunction = x.getFunction() and
  getBaseExpr = x.getBaseExpr()
select x, "getFunction:", getFunction, "getBaseExpr:", getBaseExpr
