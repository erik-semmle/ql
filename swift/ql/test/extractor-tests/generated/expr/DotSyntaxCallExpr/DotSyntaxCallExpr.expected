| dot_syntax_call.swift:6:1:6:3 | call to foo(_:_:) | getFunction: | dot_syntax_call.swift:6:3:6:3 | foo(_:_:) | getBaseExpr: | dot_syntax_call.swift:6:1:6:1 | X.Type |
| dot_syntax_call.swift:7:1:7:3 | call to bar() | getFunction: | dot_syntax_call.swift:7:3:7:3 | bar() | getBaseExpr: | dot_syntax_call.swift:7:1:7:1 | X.Type |
